### PR TITLE
SMPC+DP Comparison

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Integration SMPC+DP Comparision operations.

## How has this been tested?
Existing Trade demo tests

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/core/tensor/smpc/share_tensor.py [hunk 1]
```diff
@@ -561,33 +635,36 @@ def __eq__(self, other: Any) -> bool:
...
-        from .... import Tensor
+        # from .... import Tensor
 
-        if (
-            isinstance(self.child, Tensor)
-            and isinstance(other.child, Tensor)
-            and (self.child != other.child).child.any()  # type: ignore
-        ):
-            return False
+        # if (
```
Comment (gmuraru) [inline packages/syft/src/syft/core/tensor/smpc/share_tensor.py:642]: Q: Should those be commented?

**Review by gmuraru — APPROVED** [review]
LGTM! only one Q

File: packages/syft/src/syft/core/tensor/smpc/share_tensor.py (hunk 1 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/core/tensor/smpc/share_tensor.py:642]: That we use it as proxy george, the singleentity phitensor passes down `eq` we do not really want to process it ,we should change afterwards on the implementation

**Comment by iamtrask** [comment]
Pulling down to test now!

**Pull request merged**